nodejs/week3/meal-sharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,106 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+//...........................................Returns all meals	
+
+router.get("/", async (request, response) => {
+  const {
+    maxPrice,
+    availableReservations,
+    title,
+    createdAfter,
+    limit
+  } = request.query;
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    if (Object.keys(request.query).length === 0) {
+      const resault = await knex("meals").select("*");
+      response.json(resault);
+
+      //.............................................Get meals that has a price smaller than maxPrice	
+
+    } else if (maxPrice) {
+      let resault = await knex("meals").select("*").where("price", "<", maxPrice);
+      response.send(resault);
+    } else if (availableReservations) {
+      let resault = await knex("meals").select("*").where("maxNumberOfGuests", ">", 0);
+      response.send(resault);
+
+    } else if (title) {
+      let resault = await knex("meals").select("*").where("title", "like", `%${title}%`);
+      response.send(resault);
+
+    } else if (createdAfter) {
+      let resault = await knex("meals").select("*").where("createdAt", ">", createdAfter);
+      response.send(resault);
+
+    } else if (limit) {
+      let resault = await knex("meals").select("*").limit(limit);
+      response.send(resault);
+    }
+
+  } catch (error) {
+    throw error;
+  }
+});
+
+//...........................................Adds a new meal	
+
+router.post("/", async (request, response) => {
+
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    const titles = await knex("meals").select("*").insert(request.body)
+    response.json(titles);
+  } catch (error) {
+    throw error;
+  }
+});
+
+//..........................................Returns meal by id	
+
+router.get("/:id", async (request, response) => {
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    const titles = await knex("meals").select("*").where({
+      "id": request.params.id
+    });
+    response.json(titles);
+  } catch (error) {
+    throw error;
+  }
+});
+
+//..........................................Updates the meal by id
+
+router.put("/:id", async (request, response) => {
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    const titles = await knex("meals").select("*").where({
+      "id": request.params.id
+    }).update({
+      "price": 350
```

Comment:
Without doing anything with the request body in this `PUT`, this endpoint really doesn't serve a purpose.